src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1038,7 +1038,7 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
 
         BuildData previousBuildData = getBuildData(build.getPreviousBuild());   // read only
         BuildData buildData = copyBuildData(build.getPreviousBuild());
-        build.addAction(buildData);
```

Comment:
I suppose we could instead remove the duplicate action where I'm checking for duplication now at the end, but as best as I can tell from my testing so far, this just means a slight delay on the build data being added, and it's not added until it's in its final state, while before this it gets added lo,Inf like the previous build and then changes along the way.